enhancements/csvless-bundles.md

```diff
@@ -24,209 +24,488 @@ status: implementable
+
+### User Stories
+
+#### File Format
+
+As an Operator Author, I want a way to store my `registry+v2` bundle as set of files and directories, so that I can:
+
+- iterate on its content locally
+- use standard source control tooling to:
+  - persist and version it
+  - drive pipelines
 
-The `registry+v2` mediatype will be used to represent a bundle that does not contain a CSV. This will let OLM and other tooling differentiate this type of bundle from other types.
+##### UX
 
-### Inferring CSV Content
+A `registry+v2` bundle will be represented on-disk as a directory tree consisting of a root and two siblings:
 
-CSVs are generally ⅔s standard k8s manifests and ⅓ operator runtime information. They provide a description that can be used to gather dependencies for, install, and display an operator. When attempting to denormalize this description, it's critical that there is no loss of information capacity. To this end, it helps to split it into two categories: inferrable and non-inferrable.
+- the root is any arbitrary directory in the filesystem that contains the siblings at depth 1
+- the `manifests` directory, contains the resource manifests used to generate the final operator manifests at runtime
+  - a flat directory
+  - may only contain kubectl-able YAML streams
+- the `metadata` directory, which contains all metadata
+  - doesn't need to be flat
+  - contains metadata required for OLM to manage the operator in a `bundle.yaml` file
+  - contains an `annotations.yaml` file which contains labels to be projected onto the operator's [Image Format](#image-format).
+  - may contain additional arbitrary metadata
 
-#### Inferrable
+_Ex: Kiali Operator_
 
-To support a wider range of inferred CSV fields, manifests for resources usually generated by a CSV will need to be provided as input to `registry+v2` bundle creation:
+```sh
+$ tree kiali
+kiali
+├── manifests
+│   ├── crds.yaml
+│   ├── deployment.yaml
+│   └── rbac.yaml
+└── metadata
+    ├── annotations.yaml
+    └── bundle.yaml
+
+2 directories, 5 files
+```
 
-- Deployments for InstallStrategy
-- RBAC w/ subjects matching Deployment ServiceAccounts for Permissions and Owned, Required, and NativeAPIs
-- APIServices and Services for APIServiceDefinitions
+In order to support all of the features of a `registry+v1` bundle without a CSV, OLM must be given and/or infer the information that would have been provided by the CSV via other resources. Here are the basic requirements and inferences for `registry+v2` bundles.
 
-Required, provided, optional, and native APIs in a `registry+v2` bundle will inferred by inspecting the RBAC, CRDs, Deployments, and APIServices it contains using the following rules:
+The `manifests` directory must have at least one `Deployment` resource (to take the place of a CSV's embedded `DeploymentSpec`).
```

Comment:
Good idea.

Two questions here:
1. What does "managed" mean here?
2. Do you think we'll run into situations where "deployable" resources -- e.g. `Deployment`, `StatefulSet`, `Pod` -- are ever unmanaged.

Reply:
I thought that might be out of scope here since a CSV is still constructed. 

I think "manage" here means what OLM does for deployment in the CSV. I still don't know the full list of things we do, but I think those should all be considered before we say that we "manage" other workloads.

Reply:
> What does "managed" mean here?

Managed meaning that the object gets the full management that deployments do today in the CSV: they get stamped with a hash, recreated if the definition changes, recreated if they're deleted, etc.

Several other bundle objects are created "unmanaged" - `service` objects will be created during install but nothing will happen if they're removed after the fact. 

> Do you think we'll run into situations where "deployable" resources -- e.g. Deployment, StatefulSet, Pod -- are ever unmanaged.

I could see it being reasonable to allow that (essentially "on by default/batteries included" workloads that can then be removed).

My concern is a future where:

- We stop generating CSVs on-cluster, because we have moved on to some other representation
- We allow unmanaged deployments (/other deployable resources)
- We want to continue to support registry-v2 bundles

But now that I write this out, it probably makes more sense to do the inverse and have an "unmanaged" flag for these resources, which we can skip until we have that feature well defined.